src/sensor/pingsensor.cpp

```diff
@@ -10,7 +10,8 @@ PingSensor::PingSensor()
     ,_lostMessages(0)
 {
     _parser = new PingParserExt();
-    connect(dynamic_cast<PingParserExt*>(_parser), &PingParserExt::newMessage, this, &PingSensor::handleMessagePrivate);
+    connect(dynamic_cast<PingParserExt*>(_parser), &PingParserExt::newMessage, this, &PingSensor::handleMessagePrivate,
```

Comment:
This PR is already complex in the way that it's. This update only improves the connection and message handle speed and not the `connection` call speed itself. Since this is done in the constructor and it only happens when the device is connected, this cast time is not a huge concern right now, we can deal with it in future PRs.